lib/semian/sysv_shared_memory.rb

```diff
@@ -0,0 +1,50 @@
+module Semian
+  module SysVSharedMemory #:nodoc:
+    def self.included(base)
+      def base.do_with_sync(*names)
+        names.each do |name|
+          new_name = "#{name}_inner".freeze.to_sym
+          alias_method new_name, name
+          private new_name
+          define_method(name) do |*args, &block|
+            synchronize do
+              method(new_name).call(*args, &block)
+            end
+          end
+        end
+      end
+    end
```

Comment:
The issue is that if you define `#value` to return `@value` in `Simple`, your ability to share that with `SysV` is out since the `SysV` ones don't define `@value`. If you define `#value` to return `self.value`, well, then you have infinite looping.

I don't think the no-op synchronize works at all in this situation, is what I'm saying